### PR TITLE
Add semantic html content

### PR DESCRIPTION
**Pull request by narcissus123**
### Description

- Added new content, codes and images to the semantic html document (docs/content/html/concepts/semantic-html/semantic-html.md).
- Closes #<issue #3286>

### Type of Change

- Editing an existing entry (fixing a type, bug, issues, etc.) 

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@narcissus123 I did a first review and while I admire your enthusiasm, you've gone far beyond the issue, and have written a full-blown tutorial. Doc's isn't the place for tutorials like this. The entries are meant to be short overviews of the topic that are easily digestible. In addition, the CSS segments aren't germane to the subject of the entry. To accept this, the images (what the issue was asking for) should relate to semantic HTML, and the rest of this needs to be boiled down to a few paragraphs (three or four at most.) 

Also note we avoid first- and second-person pronouns in Docs.

**Comment by narcissus123** [comment]
@ SSwiniarski thank you, I will work on the semantic HTML document and ensure it follows the guidelines you mentioned.

**Comment by heena-patel31** [comment]
hi @SSwiniarski  Can I please make these changes?

**Comment by SSwiniarski** [comment]
@heena-patel31, this is @narcissus123's PR, it's their job to edit it. You only make edits to your own PR.

**Comment by narcissus123** [comment]
Hi @SSwiniarski, please review the latest changes I've made based on your feedback.

**Comment by SSwiniarski** [comment]
@narcissus123 The PR is better, but I am going to remove the issue link and unassign you from the issue as this does not actually address the original issue (adding media to the Semantic HTML entry). We'll be reviewing this PR on its own without regard to the original issue.

File: content/html/concepts/semantic-html/semantic-html.md [hunk 1]
```diff
@@ -44,3 +44,46 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
 - Readability: Semantic HTML also makes the website’s source code easier to read for other web developers.
 
 - SEO: It improves the website SEO, or Search Engine Optimization, which is the process of increasing the number of people that visit a webpage. With better SEO, search engines are better able to identify the content of a website and weigh the most important content appropriately.
+
+## Testing and Validating for Browser Compatibility
+
+Testing and validating for browser compatibility is essential in web development to ensure a consistent user experience. Stick to well-supported modern semantic HTML tags, checking their compatibility using resources like [Can I use](https://caniuse.com/). Additionally, employ validation tools like the [W3C Markup Validation Service](https://validator.w3.org/) to maintain code quality, identify errors, and ensure compliance with web standards, enhancing compatibility across various browsers and devices. 
+
+## Semantic HTML Best Practices: Mastering the Art
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:52]:
```suggestion
## Semantic HTML Best Practices
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 2]
```diff
@@ -44,3 +44,46 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+
+- Use Semantics for Meaning: Choose semantic HTML Tags wisely: It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:54]:
```suggestion
Use Semantics for Meaning: Choose semantic HTML Tags wisely. It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 3]
```diff
@@ -44,3 +44,46 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+  <article>
+    <!-- Main content -->
+  </article>
+  <aside>
+    <!-- Sidebar content -->
+  </aside>
+</main>
+```
+
+## Drawbacks of Using Semantic HTML: Navigating Challenges
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:85]:
```suggestion
## Drawbacks of Using Semantic HTML
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@narcissus123 I went through your edits and left a few comments.

**Comment by narcissus123** [comment]
Thank you, [SSwiniarski](https://github.com/SSwiniarski), for your review and valuable feedback. I have carefully made all the changes you suggested. Additionally, I've introduced two images to enhance the document under the "Semantic Tags" and "Use Semantics for Meaning" sections. If you have any further suggestions or if these additions are not in line with what you were looking for, please let me know, and I'll make the necessary adjustments.

File: content/html/concepts/semantic-html/semantic-html.md [hunk 4]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
 ```html
 <!-- Semantic HTML -->
 <footer>
-  <p>This is a footer</p>
+  <p>This is a footer.</p>
 </footer>
 ```
 
+![Semantic Tags](https://raw.githubusercontent.com/Codecademy/docs/main/media/<semantic-tags>.<png>)
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:40]:
```suggestion
![Semantic Tags](https://raw.githubusercontent.com/Codecademy/docs/main/media/semantic-tags.png)
```
Ad a sentence before this image describing what it shows.

File: content/html/concepts/semantic-html/semantic-html.md [hunk 5]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+
+## Testing and Validating for Browser Compatibility
+
+Testing and validating for browser compatibility is essential in web development to ensure a consistent user experience. Stick to well-supported modern semantic HTML tags, checking their compatibility using resources like [Can I use](https://caniuse.com/). Additionally, employ validation tools like the [W3C Markup Validation Service](https://validator.w3.org/) to maintain code quality, identify errors, and ensure compliance with web standards, enhancing compatibility across various browsers and devices. 
+
+## Semantic HTML Best Practices
+
+Use Semantics for Meaning: Choose semantic HTML Tags wisely: It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:56]:
```suggestion
Use Semantics for Meaning: Choose semantic HTML Tags wisely. It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 6]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+    <h2>Upcoming Tech Events</h2>
+    <p>Stay updated on the latest tech events & conferences...</p>
+    <!-- Article content here -->
+  </article>
+</section>
+```
+
+![Semantic Tag Visualization for Technology News](https://raw.githubusercontent.com/Codecademy/docs/main/media/<use-semantics-for-meaning>.<png>)
+
+Proper Nesting: Structure with intent. Proper nesting is a fundamental aspect of semantic HTML. So, it is crucial to structure content logically to reflect hierarchical relationships between semantic elements. For instance, avoid the common mistake of placing the `<aside>` element directly within the main content; `<aside>` is intended to be positioned to the side, separate from the primary content.
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:76]:
```suggestion
Proper Nesting: Proper nesting is a fundamental aspect of semantic HTML. So, it is crucial to structure content logically to reflect hierarchical relationships between semantic elements. For instance, avoid the common mistake of placing the `<aside>` element directly within the main content; `<aside>` is intended to be positioned to the side, separate from the primary content.
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 7]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+  </article>
+  <article>
+    <h2>Upcoming Tech Events</h2>
+    <p>Stay updated on the latest tech events & conferences...</p>
+    <!-- Article content here -->
+  </article>
+</section>
+```
+
+![Semantic Tag Visualization for Technology News](https://raw.githubusercontent.com/Codecademy/docs/main/media/<use-semantics-for-meaning>.<png>)
```
Comment (SSwiniarski) [inline content/html/concepts/semantic-html/semantic-html.md:74]:
```suggestion
![Semantic Tag Visualization for Technology News](https://raw.githubusercontent.com/Codecademy/docs/main/media/use-semantics-for-meaning.png)
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@narcissus123 Some more comments. I reassigned the issue as this PR now addresses it.

**Comment by narcissus123** [comment]
Hi [@SSwiniarski](https://github.com/SSwiniarski), Thank you for your feedback and for reassigning the issue to this pull request. I've made the changes based on your suggestions. Please let me know if there are any further adjustments needed.

**Review by SSwiniarski — COMMENTED** [review]
@narcissus123 looks good for a second review.

**Comment by narcissus123** [comment]
[@SSwiniarski](https://github.com/SSwiniarski), I removed the trailing space at line 52 to remove the error message provided in the "ci (lint:md)" check. But I am not sure how I can remove the other one.

File: content/html/concepts/semantic-html/semantic-html.md [hunk 8]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
 
 - Accessibility: Semantic HTML makes webpages accessible for mobile devices and for people with disabilities as well. This is because screen readers and browsers are better able to interpret the content of the site.
 
 - Readability: Semantic HTML also makes the website’s source code easier to read for other web developers.
 
 - SEO: It improves the website SEO, or Search Engine Optimization, which is the process of increasing the number of people that visit a webpage. With better SEO, search engines are better able to identify the content of a website and weigh the most important content appropriately.
+
+## Testing and Validating for Browser Compatibility
+
+Testing and validating for browser compatibility is essential in web development to ensure a consistent user experience. Stick to well-supported modern semantic HTML tags, checking their compatibility using resources like [Can I use](https://caniuse.com/). Additionally, employ validation tools like the [W3C Markup Validation Service](https://validator.w3.org/) to maintain code quality, identify errors, and ensure compliance with web standards, enhancing compatibility across various browsers and devices.
```
Comment (caupolicandiaz) [inline content/html/concepts/semantic-html/semantic-html.md:52]:
```suggestion
Testing and validating for browser compatibility is essential in web development to ensure a consistent user experience. It is best practice to employ well-supported modern semantic HTML tags, and confirming their compatibility using resources like [Can I use](https://caniuse.com/). Additionally, validation tools like the [W3C Markup Validation Service](https://validator.w3.org/) are valuable for maintaining code quality, identifying errors, and ensuring compliance with web standards. These practices can aid in supporting compatibility across various browsers and devices.
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 9]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+
+## Semantic HTML Best Practices
+
+Use Semantics for Meaning: Choose semantic HTML Tags wisely. It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```
Comment (caupolicandiaz) [inline content/html/concepts/semantic-html/semantic-html.md:56]:
```suggestion
It is important to choose tags that accurately convey the content's meaning and purpose. Avoid common mistakes like using `<article>` and `<section>` tags interchangeably. In the following example, using `<article>` for each subsection correctly represents self-contained content (news articles), while `<section>` encapsulates the broader category 'Technology News,' aligning with their intended semantic purposes.
```

File: content/html/concepts/semantic-html/semantic-html.md [hunk 10]
```diff
@@ -33,14 +33,59 @@ Semantic tags, such as `<header>` or `<footer>`, are designated for a specific p
...
+    <h2>Upcoming Tech Events</h2>
+    <p>Stay updated on the latest tech events & conferences...</p>
+    <!-- Article content here -->
+  </article>
+</section>
+```
+
+![Semantic Tag Visualization for Technology News](https://raw.githubusercontent.com/Codecademy/docs/main/media/use-semantics-for-meaning.png)
+
+Proper Nesting: Proper nesting is a fundamental aspect of semantic HTML. So, it is crucial to structure content logically to reflect hierarchical relationships between semantic elements. For instance, avoid the common mistake of placing the `<aside>` element directly within the main content; `<aside>` is intended to be positioned to the side, separate from the primary content.
```
Comment (caupolicandiaz) [inline content/html/concepts/semantic-html/semantic-html.md:76]:
```suggestion
Proper nesting is a fundamental aspect of semantic HTML. So, it is crucial to structure content logically to reflect hierarchical relationships between semantic elements. For instance, avoid the common mistake of placing the `<aside>` element directly within the main content. The `<aside>` tag is intended for content that is to be positioned to the side, or flanking, separate from the primary content.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@narcissus123 I've added a few comments. Please update the entry, and tag me when you've finished making the changes. Thanks

**Comment by narcissus123** [comment]
Hi [@caupolicandiaz](https://github.com/caupolicandiaz), I updated the entry. Please let me know if there are any further adjustments needed. Thank you.

**Review by caupolicandiaz — APPROVED** [review]
@narcissus123 @SSwiniarski This is ready to merge

**Review by SSwiniarski — APPROVED** [review]

**Pull request merged**